hera_cal/reflections.py

```diff
@@ -225,34 +157,63 @@ def model_reflections(self, dly_range, keys=None, data='clean', edgecut=0, taper
 
             # find reflection
             echo("...Modeling reflections in {}, assigning to {}".format(k, rkey), verbose=verbose)
-            (eps, amp, delays, phs, inds, sig, dfft,
-             dly_arr) = fit_reflection(data[k], dly_range, freqs, edgecut=edgecut, taper=taper,
-                                       zero_pad=zero_pad, full_freqs=self.freqs, fthin=fthin, **win_kwargs)
-            self.epsilon[rkey] = eps
-            self.amps[rkey] = amp
-            self.phs[rkey] = phs
-            self.delays[rkey] = delays
-            self.peak_ratio[rkey] = sig
-            self.dfft[k] = dfft
-
-    def write_reflections(self, output_calfits, input_calfits=None, overwrite=False, add_to_history=''):
+
+            # fit for reflection delays and amplitude
+            amp, dly, inds, sig = fit_reflection_delay(self.dfft[k], dly_range, self.delays, return_peak=False)
+
+            # make significance cut
+            if np.max(sig) < ref_sig_cut:
+                continue
+
+            # solve for phase (slowest step)
+            phs = fit_reflection_phase(self.dfft[k], dly_range, self.delays, amp, dly, fthin=fthin, Nphs=Nphs, ifft=True, ifftshift=True)
+
+            # anchor phase to 0 Hz
+            phs = (phs - 2 * np.pi * dly / 1e9 * (self.freqs[0] - zeropad * self.dnu)) % (2 * np.pi)
+
+            # form epsilon term
+            eps = construct_reflection(self.freqs, amp, dly / 1e9, phs, real=False)
+
+            self.ref_eps[rkey] = eps
+            self.ref_amp[rkey] = amp
+            self.ref_phs[rkey] = phs
+            self.ref_dly[rkey] = dly
+            self.ref_significance[rkey] = sig
+            self.ref_flags[rkey] = np.min(flags[k], axis=1, keepdims=True)
+
+        # form gains
+        self.ref_gains = form_gains(self.ref_eps)
+        for a in self.ants:
```

Comment:
I understand why you need this loop, but I find it strange. If you specify keys that don't include all autocorrelations, then the code assumes that those antennas have no reflection and reports that. It's a non-intuitive cascade of assumptions. I get why you need that, but perhaps you want to clarify in the documentation what is assumed if you don't use `keys=None`.

Reply:
Actually this was done so that when we write out gains, all antennas have a gain. Otherwise, a `calfits` without an antenna in `apply_cal` will flag the baseline when applied, which we don't want: we simply don't have a reflection for that antenna, which is fine. I'll clarify in the docs though